collated/main/A0139924W.md

```diff
@@ -0,0 +1,753 @@
+    
+    public static final String MESSAGE_EMPTY_REDO_CMD_HIST = "No more actions that can be redo.";
+
+    @Override
+    public CommandResult execute() {
+        assert model != null;
+
+        if (model.getRedoableCmdHist().size() == 0) {
+            return new CommandResult(MESSAGE_EMPTY_REDO_CMD_HIST);
+        }
+
+        UndoableCommand command = (UndoableCommand) model.getRedoableCmdHist().pop();
+        model.getUndoableCmdHist().push(command);
+        return command.redo();
+    }
+}
+```
+###### \java\tars\logic\commands\RsvCommand.java
+``` java
+     */
+    @Override
+    public CommandResult undo() {
+        if (toReserve != null) {
+            try {
+                model.deleteRsvTask(toReserve);
+                return new CommandResult(String.format(UndoCommand.MESSAGE_SUCCESS,
+                        String.format(MESSAGE_UNDO_DELETE, toReserve)));
+            } catch (RsvTaskNotFoundException e) {
+                return new CommandResult(String.format(UndoCommand.MESSAGE_UNSUCCESS,
+                        Messages.MESSAGE_RSV_TASK_CANNOT_BE_FOUND));
+            }
+        } else {
+            for (RsvTask rsvTask : rsvTasksToDelete) {
+                try {
+                    model.addRsvTask(rsvTask);
+                } catch (DuplicateTaskException e) {
+                    return new CommandResult(String.format(UndoCommand.MESSAGE_UNSUCCESS,
+                            Messages.MESSAGE_DUPLICATE_TASK));
+                }
+            }
+
+            String addedRsvTasksList = CommandResult.formatRsvTasksList(rsvTasksToDelete);
+            return new CommandResult(String.format(UndoCommand.MESSAGE_SUCCESS,
+                    String.format(MESSAGE_UNDO_ADD, addedRsvTasksList)));
+        }
+    }
+
+    /**
+```
+###### \java\tars\logic\commands\RsvCommand.java
+``` java
+     */
+    @Override
+    public CommandResult redo() {
+        if (toReserve != null) {
+            try {
+                model.addRsvTask(toReserve);
+            } catch (DuplicateTaskException e) {
+                return new CommandResult(String.format(RedoCommand.MESSAGE_UNSUCCESS,
+                        Messages.MESSAGE_DUPLICATE_TASK));
+            }
+            return new CommandResult(String.format(RedoCommand.MESSAGE_SUCCESS, String.format(MESSAGE_REDO_ADD, toReserve)));
+        } else {
+            for (RsvTask rsvTask : rsvTasksToDelete) {
+                try {
+                    model.deleteRsvTask(rsvTask);
+                } catch (RsvTaskNotFoundException e) {
+                    return new CommandResult(String.format(RedoCommand.MESSAGE_UNSUCCESS,
+                            Messages.MESSAGE_RSV_TASK_CANNOT_BE_FOUND));
+                }
+            }
+
+            String deletedRsvTasksList = CommandResult.formatRsvTasksList(rsvTasksToDelete);
+            return new CommandResult(String.format(RedoCommand.MESSAGE_SUCCESS, String.format(MESSAGE_REDO_DELETE, deletedRsvTasksList)));
```

Comment:
coding standard violation; Avoid lines longer than 80 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.